### PR TITLE
[APO-1636] Add VellumIntegrationService for native Vellum integration tools

### PR DESCRIPTION
**Pull request by alex-nork**
The purpose of this PR is to create a service that uses Vellum's native integration endpoints for tool execution.

## Changes Made
- Add VellumIntegrationService class that uses native Vellum client SDK endpoints
- Implement get_tool_definition() and execute_tool() methods
- Add comprehensive unit tests

## Testing
- Unit tests added for all service methods
- Tests cover success cases, error handling, and edge cases
- All tests passing

---
*This PR was created with Claude Code*

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ tsconfig.tsbuildinfo
 .idea
 .DS_Store
 htmlcov/
+CLAUDE.md
```
Comment (alex-nork) [inline .gitignore:16]: not sure if we want to track this so ignoring for now

**Review by NgoHarrison — COMMENTED** [review]
✅

File: src/vellum/workflows/integrations/tests/test_vellum_integration_service.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+from vellum.workflows.exceptions import NodeException
+from vellum.workflows.integrations.vellum_integration_service import VellumIntegrationService
+
+
+def test_vellum_integration_service_get_tool_definition_success():
+    """Test that tool definitions are successfully retrieved from Vellum API"""
+    with mock.patch(
+        "vellum.workflows.integrations.vellum_integration_service.create_vellum_client"
+    ) as mock_create_client:
```
Comment (dvargasfuertes) [inline src/vellum/workflows/integrations/tests/test_vellum_integration_service.py:12]: [nit] we have a `vellum_client` fixture we could reuse for tests like these

File: src/vellum/workflows/integrations/vellum_integration_service.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            NodeException: If the tool definition cannot be retrieved
+        """
+        try:
+            response = self._client.integrations.retrieve_integration_tool_definition(
+                integration=integration,
+                provider=provider,
+                tool_name=tool_name,
+            )
+
+            # Convert the response to a dict format matching what's expected
```
Comment (dvargasfuertes) [inline src/vellum/workflows/integrations/vellum_integration_service.py:46]: [nit] `format matching what's expected` <- if we care about that, would prefer returning a typed pydantic model instead of a dict

**Review by dvargas92495 — COMMENTED** [review]
✅

**Comment by alex-nork** [comment]
@codex address this feedback
